File: Examples/Node.js/SDK/Oleobjects/ConvertOleObjectToImage.js
// ExStart:1
var fs = require('fs');
var assert = require('assert');
var StorageApi = require('asposestoragecloud');
var CellsApi = require('asposecellscloud');
var configProps = require('../Config/config.json');
var data_path = '../../../../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
// Instantiate Aspose Cells API SDK
var cellsApi = new CellsApi(config);

// Set input file name
var filename = "Sample_OleObject_Book1";
var name = filename + ".xlsx";
var sheetName = "Sheet1";
var objectNumber = 0;
var format = "png";
try {
// Upload source file to aspose cloud storage
	storageApi.PutCreate(name, null, null, data_path + name , function(responseMessage) {
		assert.equal(responseMessage.status, 'OK');
		
		// Invoke Aspose.Cells Cloud SDK API to convert an OLE object to image
		cellsApi.GetWorksheetOleObjectWithFormat(name, sheetName, objectNumber, format, null, null, function(responseMessage) {
			assert.equal(responseMessage.status, 'OK');
			
			// Get converted image from response stream
			var outfilename = filename + objectNumber + "." + format;
			var writeStream = fs.createWriteStream(data_path + "_out_" + outfilename);
			writeStream.write(responseMessage.body);

			});

		});

}catch (e) {
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1

File: Examples/Python/ConvertOLEObjectToImage.py
import asposecellscloud
from asposecellscloud.CellsApi import CellsApi
from asposecellscloud.CellsApi import ApiException

import asposestoragecloud
from asposestoragecloud.StorageApi import StorageApi

apiKey = "XXXXX" #sepcify App Key
appSid = "XXXXX" #sepcify App SID
apiServer = "http://api.aspose.com/v1.1"
data_folder = "../../data/"

#Instantiate Aspose Storage API SDK
storage_apiClient = asposestoragecloud.ApiClient.ApiClient(apiKey, appSid, True)
storageApi = StorageApi(storage_apiClient)
#Instantiate Aspose Cells API SDK
api_client = asposecellscloud.ApiClient.ApiClient(apiKey, appSid, True)
cellsApi = CellsApi(api_client);

#set input file name
name = "Sample_OleObject_Book1"
filename = name + ".xlsx"
sheetName = "Sheet1"
objectNumber = 0
format = "png"

#upload file to aspose cloud storage
storageApi.PutCreate(Path=filename, file=data_folder + filename)

try:
    #invoke Aspose.Cells Cloud SDK API to convert an OLE object to image
    response = cellsApi.GetWorksheetOleObjectWithFormat(name=filename, sheetName=sheetName, objectNumber=objectNumber, format=format)

    if response.Status == "OK":
        #get converted image from response stream
        outfilename = "c:/temp/" +  name + str(objectNumber) + "." +  format
        with open(outfilename, 'wb') as f:
                    for chunk in response.InputStream:
                        f.write(chunk)

except ApiException as ex:
            print "ApiException:"
            print "Code:" + str(ex.code)
            print "Message:" + ex.message
